### PR TITLE
Add Some Role Manipulation

### PR DESCRIPTION
**Pull request by Kashalls**
Make sure to add the following to your defaultGuildSchema!
`     	.add('selfroles', folder => folder
            .add('roles', 'role', { array: true })
            .add('enabled', 'boolean', { default: true }))`

This PR adds the following commands:
- roleme : Used to add, remove or list roles that were made available by the managerole command.
- toggleroles : Used to toggle self-assignable roles on or off.
- managerole : Used to toggle a role on or off, code be configured to allow only one role and/or disallow roles with "Moderation" permissions.

### CODE REVIEW

**Comment by kyranet** [comment]
CI is red, can you fix the errors, please?

**Comment by Kashalls** [comment]
@kyranet It has been fixed

File: commands/General/Role Manipulation/manageroles.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+	    .add('roles', 'role', { array: true })
+            .add('enabled', 'boolean', { default: true }))
+*/
+
+module.exports = class extends Command {
+
+	constructor(...args) {
+		super(...args, {
+			runIn: ['text'],
+			requiredPermissions: ['SEND_MESSAGES', 'EMBED_LINKS', 'USE_EXTERNAL_EMOJIS', 'MANAGE_ROLES'],
```
Comment (KingDGrizzle) [inline commands/General/Role Manipulation/manageroles.js]: `SEND_MESSAGE` is generic and already checked in the monitor, so not needed.
`USE_EXTERNAL_EMOJIS`? Is that _really_ needed?

File: commands/General/Role Manipulation/manageroles.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+		const assigned = await msg.member.roles.remove(role, 'Self Deassigned').catch(() => null);
+		if (!assigned) return msg.send(`｢ **Error** ｣ There was an error, please try again later.`);
+		return msg.send(`｢ **RoleMe** ｣ The role has been deassigned.`);
+	}
+
+	roleCheck(msg) {
+		const memberRoles = msg.member.roles.map(roleRole => roleRole.id);
+		const { roles } = msg.guild.settings.selfroles;
+
+		var ret = [];
```
Comment (KingDGrizzle) [inline commands/General/Role Manipulation/manageroles.js]: Please use const here

File: commands/General/Role Manipulation/manageroles.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+		for (var i in memberRoles) {
```
Comment (KingDGrizzle) [inline commands/General/Role Manipulation/manageroles.js]:
```js
for (const roleID of memberRoles)
    if (roles.includes(roleID)) ret.push(roleID);

return ret.length >= 1;
```

File: commands/General/Role Manipulation/roleme.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+		.add('roles', 'role', { array: true })
+            	.add('enabled', 'boolean', { default: true }))
+*/
+
+module.exports = class extends Command {
+
+	constructor(...args) {
+		super(...args, {
+			runIn: ['text'],
+			requiredPermissions: ['SEND_MESSAGES', 'EMBED_LINKS', 'USE_EXTERNAL_EMOJIS', 'MANAGE_ROLES'],
```
Comment (KingDGrizzle) [inline commands/General/Role Manipulation/roleme.js]: Ditto

File: commands/General/Role Manipulation/roleme.js [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+const { RichDisplay } = require('klasa');
+const { MessageEmbed } = require('discord.js');
+
+/* You need to add the following lines to your guild schematic manager:
+     	.add('selfroles', folder => folder
+		.add('roles', 'role', { array: true })
+            	.add('enabled', 'boolean', { default: true }))
+*/
+
+module.exports = class extends Command {
```
Comment (KingDGrizzle) [inline commands/General/Role Manipulation/roleme.js]: This command looks like a carbon copy of manageroles.js up above... And from the file names, I assume that's unintended.

File: commands/General/Role Manipulation/toggleroles.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+			extendedHelp: 'No extended help available.'
+		});
+	}
+
+	async run(msg) {
+		const current = msg.guild.settings.get('selfroles.enabled');
+		msg.guild.settings.update('selfroles.enabled', !current);
+		if (current) {
+			return msg.send(`｢ **Self Roles** ｣ Self Roles have been globally disabled. Happy Hunting for Inertia!`);
+		} else {
```
Comment (KingDGrizzle) [inline commands/General/Role Manipulation/toggleroles.js]: Useless else

Comment (tech6hutch) [reply commands/General/Role Manipulation/toggleroles.js]: I'd be tempted to do
```js
return msg.send(`｢ **Self Roles** ｣ Self Roles have been globally ${current ?
    'disabled. Happy Hunting for Inertia!' :
    "enabled. Seems like we got 'em! :)"}`);
```

File: commands/General/Role Manipulation/manageroles.js (hunk 3 above)
Comment (tech6hutch) [reply commands/General/Role Manipulation/manageroles.js]: I'm pretty sure ESLint will complain about that unless there are curlies around that for block or it's put on one line

**Comment by kyranet** [comment]
Bump, requested changes a long time ago.

File: commands/General/Role Manipulation/toggleroles.js (hunk 6 above)
Comment (tech6hutch) [reply commands/General/Role Manipulation/toggleroles.js]: Also there's an inconsistency here: in one place they're referred to in the plural, in another place, in the singular.

**Comment by Kashalls** [comment]
Thanks for reminder, will fix asap.

**Comment by kyranet** [comment]
Closing due to the age of the requested changes, feel free to make a new PR later.

**Pull request closed without merging**